README.md

```diff
@@ -896,6 +901,81 @@ following articles:
 - for [IntelliJ](https://www.jetbrains.com/idea/help/run-debug-configuration-remote.html)
 
+## Docker
+
+### Build Emissary Docker Image
+Maven can be used to create the docker image. There is a profile that was created to run the docker image build that, by default,
+has been turned off. We'll need to add the docker profile, along with the dist profile, to trigger an assembly. From the
+project root, run the following maven command:
+```
+mvn clean package -Pdist,docker
+```
+
+Alternatively, we can use Docker directly. First run a full maven build and then run the ```docker build``` command:
+```
+mvn clean package -Pdist
+docker build . -t emissary:latest
+```
+
+### Run Emissary with Docker
+Once the image is successfully built, the image should be in your list of local images. Run ```docker images``` and there
+should be an entry for REPOSITORY:emissary and TAG:latest:
+```
+[~]$ docker images
+   REPOSITORY          TAG                 IMAGE ID            CREATED             SIZE
+   emissary            latest              e740d5f23a79        44 hours ago        620MB
+   centos              7                   49f7960eb7e4        2 days ago          200MB
+```
+
+To run files through Emissary, we'll need to volume mount local directories into the running container. Let's create two local directories for 
+input/output to Emissary:
+```
+mkdir input
+mkdir output
+```
+
+Now that we have two target directories, we can use the ```-v``` option to mount them into the container. To start Emissary, run the sample command:
+```
+docker run -it --rm -v ${PWD}/input:/opt/emissary/target/data:Z -v ${PWD}/output:/opt/emissary/localoutput:Z --name emissary emissary
+```
+
+Once Emissary starts up, we should see a log line that says: "Started EmissaryServer at http://localhost:8001." We now can copy files into 
+```input/InputData/``` for Emissary to process. When the processing has finish, the files will be moved to ```input/DoneData```. All extracted 
+content can be found in the local ```output``` directory. Depending on your specific configuration, there should be output in the ```output/json``` 
+directory.
+
+To monitor Emissary in a container, we need to connect to the running container. If we want to run the agents command,
+we simply need to run the following command:
+```
+docker exec -it emissary ./emissary agents -p 8001 --mon
+```
+
+To see the environment settings:
+```
+docker exec -it emissary ./emissary env
+```
+
+If you want to connect to the web front-end for Emissary, some additional steps are need to configure jetty. We need to
+give the container a hostname and pass that onto Emissary from the command line. Sample command:
+```
+docker run -it --rm --name emissary --hostname emissary-001 -p 8001:8001 emissary server -a 2 -p 8001 -s http -h emissary-001
+```
+
+Then from a browser, go to http://localhost:8001/ to see the endpoints.
```

Comment:
This didn't work for me. Do we need a different URL?

Reply:
I'm pretty sure this worked for me

Reply:
I still can't get to it. I am running this on an EC2 instance and then I am trying to access it through a firefox browser that is being X11 forwarded back from my EC2 instance. I get an "Unable to connect" error from firefox. Was there anything else additional you needed to do in order to enable this?

Reply:
This sounds like an issue running a Docker container with X11 forwarding, not specific to this Docker image. I am running it locally on my box, so localhost:8001 is working fine for me. Unless there is a issue with the image itself, I'm marking this comment resolved.